lib/src/context.dart

```diff
@@ -0,0 +1,147 @@
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+@JS()
+library react_dart_context;
+
+import 'dart:js_util';
+
+import 'package:js/js.dart';
+import 'package:react/react_client/react_interop.dart';
+import 'package:react/react_client.dart';
+
+/// The return type of [createContext], Wraps [ReactContext] for use in Dart.
+/// Allows access to [Provider] and [Consumer] Components.
+///
+/// __Should not be instantiated without using [createContext]__
+///
+/// __Example__:
+///
+///     Context MyContext = createContext('test');
+///
+///     class MyContextTypeClass extends react.Component2 {
+///       @override
+///       final contextType = MyContext;
+///
+///       render() {
+///         return react.span({}, [
+///           '${this.context}', // Outputs: 'test'
+///         ]);
+///       }
+///     }
+///
+/// // OR
+///
+///     Context MyContext = createContext();
+///
+///     class MyClass extends react.Component2 {
+///       render() {
+///         return MyContext.Provider({'value': 'new context value'}, [
+///           MyContext.Consumer({}, (value) {
+///             return react.span({}, [
+///               '$value', // Outputs: 'new context value'
+///             ]),
+///           });
+///         ]);
+///       }
+///     }
+///
+/// Learn more at: https://reactjs.org/docs/context.html
+class Context {
+  Context(this.Provider, this.Consumer, this._jsThis);
+  final ReactContext _jsThis;
+
+  /// Every [Context] object comes with a Provider component that allows consuming components to subscribe
+  /// to context changes.
+  ///
+  /// Accepts a `value` prop to be passed to consuming components that are descendants of this [Provider].
+  final ReactJsContextComponentFactoryProxy Provider;
+
+  /// A React component that subscribes to context changes.
+  /// Requires a function as a child. The function receives the current context value and returns a React node.
+  final ReactJsContextComponentFactoryProxy Consumer;
+  ReactContext get jsThis => _jsThis;
+}
+
+/// Creates a [Context] object. When React renders a component that subscribes to this [Context]
+/// object it will read the current context value from the closest matching Provider above it in the tree.
+///
+/// The `defaultValue` argument is only used when a component does not have a matching [Context.Provider]
+/// above it in the tree. This can be helpful for testing components in isolation without wrapping them.
+///
+/// __Example__:
+///
+///     Context MyContext = createContext('test');
+///
+///     class MyContextTypeClass extends react.Component2 {
+///       @override
+///       final contextType = MyContext;
+///
+///       render() {
+///         return react.span({}, [
+///           '${this.context}', // Outputs: 'test'
+///         ]);
+///       }
+///     }
+///
+/// ___ OR ___
+///
+///     Context MyContext = createContext();
```

Comment:
#nit this should have the react prefix
```suggestion
///     react.Context MyContext = react.createContext();
```